readthedocs/core/unresolver.py

```diff
@@ -141,46 +170,61 @@ def unresolve(self, url, append_indexhtml=True):
         :param append_indexhtml: If `True` directories will be normalized
          to end with ``/index.html``.
         """
-        parsed = urlparse(url)
-        domain = self.get_domain_from_host(parsed.netloc)
+        parsed_url = urlparse(url)
+        domain = self.get_domain_from_host(parsed_url.netloc)
         unresolved_domain = self.unresolve_domain(domain)
+        return self._unresolve(
+            unresolved_domain=unresolved_domain,
+            parsed_url=parsed_url,
+            append_indexhtml=append_indexhtml,
+        )
+
+    def unresolve_path(self, unresolved_domain, path, append_indexhtml=True):
+        """
+        Unresolve a path given a unresolved domain.
 
+        This is the same as the unresolve method,
+        but this method takes an unresolved domain
+        from unresolve_domain as input.
+
+        :param unresolved_domain: An UnresolvedDomain object.
+        :param path: Path to unresolve (this shouldn't include the protocol or querystrings).
+        :param append_indexhtml: If `True` directories will be normalized
+         to end with ``/index.html``.
+        """
+        # We don't call unparse() on the path,
+        # since it could be parsed as a full URL if it starts with a protocol.
+        parsed_url = ParseResult(
+            scheme="", netloc="", path=path, params="", query="", fragment=""
+        )
```

Comment:
Shouldn't we validate that `path` comes in the exact shape we require here?

Reply:
Not sure if I understand what you mean with shape, anything can be a path, this is the path as we received from proxito.

Reply:
We are expecting a string with a particular structure. For example, it can't be `https://humitos.dev/page/`, it _has to be_ just `/page/`. I'm saying that we should validate the `path` to be what we expect to be.

Reply:
what we expect is a random string

Reply:
In the code comment you said that it should not start with a protocol, tho. I'm confused what are the correct values for `path` then. Can you expand on this?

Reply:
The comment says we can't use `unparse()` because the path could start with a protocol, or anything really that can confuse `unparse()` so we manually build the `ParseResult` class.

Reply:
Yeah, I got that. I'm not saying we have to use `unparse()`. I'm saying that we can _validate_ the string to be exactly in the shape we need to be.